### PR TITLE
KAFKA-3934: Start scripts enable GC by default with no way t…

### PR DESCRIPTION
**Pull request by granthenke**
…o disable

### CODE REVIEW

**Comment by granthenke** [comment]
I discuss this a bit in the jira ([KAFKA-3934](https://issues.apache.org/jira/browse/KAFKA-3934)) but would like to bring it up here too. 

I am not sure why the existing scripts check for variables with the style:

``` bash
if [ "x$FOO" = "x" ]; then
   export FOO="$FOO_DEFAULT"
fi
```

Instead of something like:

``` bash
export FOO=${FOO-"$FOO_DEFAULT"} 
```

The first example means unset and "" mean the same thing. I am not sure that is favorable behavior, but even that can be achieved in the shorter format. All varieties are covered in this [reference table](http://i.stack.imgur.com/T2Fp8.png).

**Comment by ijuma** [comment]
Thoughts @granders?

File: bin/kafka-server-start.sh [hunk 1]
```diff
@@ -29,7 +29,7 @@ if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
     export KAFKA_HEAP_OPTS="-Xmx1G -Xms1G"
 fi
 
-EXTRA_ARGS="-name kafkaServer -loggc"
+export EXTRA_ARGS=${EXTRA_ARGS-'-name kafkaServer -loggc'}
```
Comment (ewencp) [inline bin/kafka-server-start.sh]: Why was `export` added? The quotes also seem to be unnecessary, but I guess they don't hurt.

**Comment by ewencp** [comment]
@granthenke I think different people just get used to handling different subsets of sh, and it's often easier to go the more verbose route than make sure you're not using a bashism in a sh script (although we decided at some point that assuming bash is fine as long as we use it in the shebang). For example, it seems like the form you used is valid sh, but the forms `${FOO=default}` and `${FOO:=default}` are bashisms.

**Comment by granthenke** [comment]
@ewencp That makes sense. I could stick with the existing style if you think there is a benefit in consistency. I could even move the other checks over to this style. Or I can leave the patch style as is. Any preference?

**Comment by ewencp** [comment]
@granthenke I'd probably just leave both as they are for now. There is actually a difference between them -- the `if` version will override variables if they exist but are empty, whereas the version you're using only uses the default for unset variables. So, e.g., if you wanted `EXTRA_ARGS=""`, you'd need to use your version (or know why it is being overridden and add whitespace or something). So probably your version is the right way to handle most args like this, although I'm not sure its worth the effort to proactively clean up the rest of those cases.

Removing the `export` is important though -- you should really only `export` if it should be visible to subprocesses, but in this case its just a local variable.

**Comment by granthenke** [comment]
@ewencp Thanks for the review. I removed the export.

**Comment by ewencp** [comment]
@granthenke Just noticed that we have the same in `zookeeper-server-start.sh`, shall we just fix both at the same time?

**Comment by granthenke** [comment]
@ewencp Happy to fix both. Will update shortly.

**Comment by ewencp** [comment]
LGTM, thanks!

**Pull request closed without merging**